ha_cluster_exporter.go

```diff
@@ -183,34 +184,36 @@ func main() {
 
 	// set SBD device metrics
 	go func() {
-		for {
-			log.Println("[INFO]: Reading cluster SBD configuration..")
-			// read configuration of SBD
-			sbdConfiguration, err := readSdbFile()
-			if err != nil {
-				log.Println(err)
-				continue
-			}
-			// retrieve a list of sbd devices
-			sbdDevices := getSbdDevices(sbdConfiguration)
-			// set and return a map of sbd devices with true healthy, false not
-			sbdStatus := setSbdDeviceHealth(sbdDevices)
-
-			if len(sbdStatus) == 0 {
-				log.Println("[WARN]: Could not retrieve any sbd device")
-				continue
-			}
+		if _, err := os.Stat("/etc/sysconfig/sbd"); os.IsExist(err) {
+			for {
+				log.Println("[INFO]: Reading cluster SBD configuration..")
+				// read configuration of SBD
+				sbdConfiguration, err := readSdbFile()
+				if err != nil {
+					log.Println(err)
+					continue
+				}
+				// retrieve a list of sbd devices
+				sbdDevices := getSbdDevices(sbdConfiguration)
+				// set and return a map of sbd devices with true healthy, false not
+				sbdStatus := setSbdDeviceHealth(sbdDevices)
+
+				if len(sbdStatus) == 0 {
+					log.Println("[WARN]: Could not retrieve any sbd device")
+					continue
+				}
 
-			for sbdDev, sbdStatusBool := range sbdStatus {
-				// true it means the sbd device is healthy
-				if sbdStatusBool == true {
-					sbdDevStatus.WithLabelValues(sbdDev).Set(float64(1))
-				} else {
-					sbdDevStatus.WithLabelValues(sbdDev).Set(float64(0))
+				for sbdDev, sbdStatusBool := range sbdStatus {
+					// true it means the sbd device is healthy
+					if sbdStatusBool == true {
+						sbdDevStatus.WithLabelValues(sbdDev).Set(float64(1))
+					} else {
+						sbdDevStatus.WithLabelValues(sbdDev).Set(float64(0))
+					}
 				}
-			}
 
-			time.Sleep(time.Duration(int64(*timeoutSeconds)) * time.Second)
+				time.Sleep(time.Duration(int64(*timeoutSeconds)) * time.Second)
+			}
```

Comment:
rather than indenting the whole `for` loop, can I suggest the "fail fast, fail early" approach? I.e. just add an if with inverted conditional logic and return out of the goroutine early if the condition is not met.

Reply:
Made this change, :+1:

Reply:
yes good catch! That is a guard clause approach which make more sense

Reply:
did you tested manually for any case? should look good to me anyways thx ! :sun_with_face:

Reply:
@MalloZup I'll add some unit tests in my refactoring.